src/OrchardCore.Modules/OrchardCore.Workflows/Controllers/WorkflowTypeController.cs

```diff
@@ -125,10 +125,18 @@ public async Task<IActionResult> Index(WorkflowTypeIndexOptions options, PagerPa
                 .ListAsync();
 
             var workflowTypeIds = workflowTypes.Select(x => x.WorkflowTypeId).ToList();
-            var workflowGroups = (await _session.QueryIndex<WorkflowIndex>(x => x.WorkflowTypeId.IsIn(workflowTypeIds))
-                .ListAsync())
-                .GroupBy(x => x.WorkflowTypeId)
-                .ToDictionary(x => x.Key);
+
+            // Changing the workflowTypeIds list to array is needed to be able to modify the original list while going
+            // through it.
+            foreach (var workflowTypeId in workflowTypeIds.ToArray())
```

Comment:
Could you try this one

```csharp
var workflowInstances = await _session.QueryIndex<WorkflowIndex>(index => index.WorkflowTypeId.IsNotIn(workflowTypeIds))

workflowTypeIds.RemoveAll(id => workflowInstances.Any(workflow => workflow.WorkflowTypeId == id));
```

Reply:
I already tried it, you can read the result here: [https://github.com/OrchardCMS/OrchardCore/pull/13680/files#r1205826624](https://github.com/OrchardCMS/OrchardCore/pull/13680/files#r1205826624)

Note that the original query that has a performance problem is:
`await _session.QueryIndex<WorkflowIndex>(x => x.WorkflowTypeId.IsIn(workflowTypeIds)`
So negating the logic won't solve the problem.

Reply:
Please request a review from Seb ou PERF guru :)

Reply:
Could you check this PR please? @sebastienros